lib/promiscuous_black_hole/operation.rb

```diff
@@ -60,7 +61,7 @@ def destroy
     end
 
     def persist_embedded_records
-      StaleEmbeddingsDestroyer.new(message.table_name, message.id).process
+      StaleEmbeddingsDestroyer.new(message.table_name, message.id).process unless @embedded
```

Comment:
it runs recursively, so you only want to run it once from the non-embedded parent document. kind of a weird way for it to work, but it makes the caching really easy